### PR TITLE
5424 setup typescript (Work in progress, do not merge)

### PR DESCRIPTION
**Pull request by erwinheitzman**
## Proposed changes

As discussed in  #5424 we want to move the project from Babel to TypeScript.
Let's discuss the changes made and see where to go from here.

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

### CODE REVIEW

**Comment by mathew-jithin** [comment]
@christian-bromann @erwinheitzman Was trying to add in a new package (service). I think as Typescript adoption is being done, is this the same format/styling that needs to be followed?

**Comment by christian-bromann** [comment]
@mathew-jithin please create and release this new service under your user name until we have figured out this transition. We need to be careful adding new packages to the mono repo as the build time is already huge.

**Comment by mathew-jithin** [comment]
@christian-bromann Sure, will do the same.

**Comment by erwinheitzman** [comment]
Pushed a change which makes use of `lerna run tsc` instead of relying on TS alone, much better :)

**Comment by christian-bromann** [comment]
@erwinheitzman what is the current state of this? Anything I can do to help?

File: packages/wdio-runner/src/index.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ export default class Runner extends EventEmitter {
         }
 
         initialiseWorkerService(this.config, caps, args.ignoredWorkerServices)
-            .map(::this.configParser.addService)
+            .map(this.configParser.addService.bind(this))
```
Comment (christian-bromann) [inline packages/wdio-runner/src/index.js:77]:
```suggestion
            .map(this.configParser.addService.bind(this.configParser))
```

File: packages/wdio-utils/src/shim.js [hunk 2]
```diff
@@ -127,7 +129,9 @@ let executeSync = async function (fn, retries, args = []) {
  * @param  {Array}    args       arguments passed to hook
  * @return {Promise}             that gets resolved once test/hook is done or was retried enough
  */
-const executeAsync = async function (fn, retries, args = []) {
+let executeAsync
+
+executeAsync = async function executeAsync(fn, retries, args = []) {
```
Comment (christian-bromann) [inline packages/wdio-utils/src/shim.js:134]: this can stay a `const`

**Comment by christian-bromann** [comment]
ping @erwinheitzman

**Comment by erwinheitzman** [comment]
Should be good now 🥳

File: packages/wdio-devtools-service/src/index.js [hunk 3]
```diff
@@ -152,10 +152,10 @@ export default class DevToolsService {
         this.commandHandler = new CommandHandler(this.session, this.page)
         this.traceGatherer = new TraceGatherer(this.puppeteer, this.session, this.page)
 
-        this.session.on('Page.loadEventFired', :: this.traceGatherer.onLoadEventFired)
-        this.session.on('Page.frameNavigated', :: this.traceGatherer.onFrameNavigated)
+        this.session.on('Page.loadEventFired', this.traceGatherer.onLoadEventFired)
+        this.session.on('Page.frameNavigated', this.traceGatherer.onFrameNavigated)
 
-        this.page.on('requestfailed', :: this.traceGatherer.onFrameLoadFail)
+        this.page.on('requestfailed', this.traceGatherer.onFrameLoadFail)
```
Comment (christian-bromann) [inline packages/wdio-devtools-service/src/index.js:158]:
```suggestion
        this.page.on('requestfailed', this.traceGatherer.onFrameLoadFail.bind(this.traceGatherer))
```

File: packages/wdio-devtools-service/src/index.js [hunk 4]
```diff
@@ -175,8 +175,8 @@ export default class DevToolsService {
             global.browser.emit(method, data.params)
         })
 
-        global.browser.addCommand('enablePerformanceAudits', :: this._enablePerformanceAudits)
-        global.browser.addCommand('disablePerformanceAudits', :: this._disablePerformanceAudits)
-        global.browser.addCommand('emulateDevice', :: this._emulateDevice)
+        global.browser.addCommand('enablePerformanceAudits', this._enablePerformanceAudits)
+        global.browser.addCommand('disablePerformanceAudits', this._disablePerformanceAudits)
+        global.browser.addCommand('emulateDevice', this._emulateDevice)
```
Comment (christian-bromann) [inline packages/wdio-devtools-service/src/index.js:180]:
```suggestion
        global.browser.addCommand('enablePerformanceAudits', this._enablePerformanceAudits.bind(this))
        global.browser.addCommand('disablePerformanceAudits', this._disablePerformanceAudits.bind(this))
        global.browser.addCommand('emulateDevice', this._emulateDevice.bind(this))
```

File: packages/wdio-local-runner/src/run.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ import { SHUTDOWN_TIMEOUT } from './constants'
 const log = logger('@wdio/local-runner')
 
 export const runner = new Runner()
-runner.on('exit', ::process.exit)
+runner.on('exit', process.exit)
```
Comment (christian-bromann) [inline packages/wdio-local-runner/src/run.js:11]:
```suggestion
runner.on('exit', process.exit.bind(process))
```

File: packages/wdio-utils/src/shim.js [hunk 6]
```diff
@@ -94,7 +94,9 @@ let wrapCommand = function wrapCommand (commandName, fn) {
  * @param  {Array}    args       arguments passed to hook
  * @return {Promise}             that gets resolved once test/hook is done or was retried enough
  */
-let executeSync = async function (fn, retries, args = []) {
+let executeSync
+
+executeSync = async function executeSync(fn, retries, args = []) {
```
Comment (christian-bromann) [inline packages/wdio-utils/src/shim.js:99]:
```suggestion
let executeSync = async function executeSync(fn, retries, args = []) {
```

**Review by christian-bromann — COMMENTED** [review]
a couple more findings, after that we are good to go

**Review by christian-bromann — APPROVED** [review]
👏 nice work!

**Pull request merged**